base/number.jl

```diff
@@ -133,11 +133,41 @@ zero{T<:Number}(::Type{T}) = convert(T,0)
 """
     one(x)
 
-Get the multiplicative identity element for the type of `x` (`x` can also specify the type
-itself). For matrices, returns an identity matrix of the appropriate size and type.
+Return a multiplicative identity for `x` (which can be either a value
+or a type), of the same shape and numeric precision as `x`.  (For matrices,
```

Comment:
I didn't get the sense that there was any kind of consensus on whether `one(matrix)` should stop returning a matrix.  From an algebraic perspective, returning a matrix makes more sense.

Moreover, `oneunit(x)` only makes sense for matrices if `one` returns a matrix, since for `oneunit(x::T)` you have to be able to do `T(one(x))`, and this will fail if `T::Matrix` and `one(x)::Number`.